modules/sops/templates/default.nix

```
@@ -0,0 +1,103 @@
+{ config, pkgs, lib, options, ... }:
+with lib;
+with lib.types;
+with builtins;
+let
+  cfg = config.sops;
+  secretsForUsers = lib.filterAttrs (_: v: v.neededForUsers) cfg.secrets;
+  users = config.users.users;
+  substitute = pkgs.writers.writePython3 "substitute" { } (readFile ./subs.py);
+  subst-pairs = pkgs.writeText "pairs" (concatMapStringsSep "\n" (name:
+    "${toString config.sops.placeholder.${name}} ${
+      config.sops.secrets.${name}.path
+    }") (attrNames config.sops.secrets));
+  coercibleToString = mkOptionType {
+    name = "coercibleToString";
+    description = "value that can be coerced to string";
+    check = strings.isCoercibleToString;
+    merge = mergeEqualOption;
+  };
+  templateType = submodule ({ config, ... }: {
+    options = {
+      name = mkOption {
+        type = str;
+        default = config._module.args.name;
+        description = ''
+          Name of the file used in /run/secrets-rendered
+        '';
+      };
+      path = mkOption {
+        type = str;
+        default = "/run/secrets-rendered/${config.name}";
+      };
+      content = mkOption {
+        type = lines;
+        default = "";
+        description = ''
+          Content of the file
+        '';
+      };
+      mode = mkOption {
+        type = str;
+        default = "0400";
+        description = ''
+          Permissions mode of the rendered secret file in octal.
+        '';
+      };
+      owner = mkOption {
+        type = str;
+        default = "root";
+        description = ''
+          User of the file.
+        '';
+      };
+      group = mkOption {
+        type = str;
+        default = users.${config.owner}.group;
```

```suggestion
        default = config.users.users.${config.owner}.group;
```
we don't need to clutter the scope if it is only used once

---

The `config` in submodule shadows the toplevel `config`, so this is actually needed.